sklearn/utils/tests/test_cython_templating.py

```diff
@@ -0,0 +1,15 @@
+import pathlib
+import pytest
+import sklearn
+
+TEMPITA_EXTENSION = "tp"
+
+
+def test_files_generated_by_templates_are_git_ignored():
+    gitignore_file = pathlib.Path(sklearn.__file__).parent.parent / ".gitignore"
+    if not gitignore_file.exists():
+        pytest.skip("Tests are not run from the source folder")
+    base_dir = pathlib.Path(sklearn.__file__).parent
+    for filename in base_dir.glob(f"**/*.{TEMPITA_EXTENSION}"):
+        filename_wo_tempita_suffix = filename.with_suffix("")
+        assert not filename_wo_tempita_suffix.exists()
```

Comment:
This test is not testing if generated cython files are git-ignored but rather if they do not exist on the file system (this makes the CI fail at the moment because they exist).

You should check against the content of the `.gitignore` file as you started with defining `gitignore_file`.

Also this test assumes that the generated source have the same filename up extensions.
This is the natural and current behavior to expect which is implemented in `gen_from_templates`:

https://github.com/scikit-learn/scikit-learn/blob/82df48934eba1df9a1ed3be98aaace8eada59e6e/sklearn/_build_utils/__init__.py#L83-L103

IMO, `gen_from_templates`' behavior should be tested if enlarging the scope of this PR.

Reply:
I think it's enough to just to test the gitignore consistency for this PR.